### ISSUE TITLE
Nima should have http connection configurable per socket 

### ISSUE THREAD

**Issue by danielkec** [helidon-io/helidon#5967]
It is possible with Helidon SE:
```java
WebServer.builder()
                .routing(() -> routing)
                .continueImmediately(false)
                .addSocket(SocketConfiguration.builder()
                        .name(CONTINUE_IMMEDIATELY_SOCKET)
                        .continueImmediately(true)
                        .build(), routing)
                .build()
                .start()
                .await(TEST_TIMEOUT);
``` 
but with Nima only per protocol configuration is possible:
```java
        WebServer.builder()
                .addConnectionProvider(Http1ConnectionProvider.builder()
                        .http1Config(DefaultHttp1Config.builder()
                                .continueImmediately(true)
                                .build())
                        .build())
                .routing(...)
                .build();
```

**Comment by tomas-langer** [comment]
After a bit of refactoring of WebServer configuration, this is now possible:
```
WebServer.builder()
                .putSocket("socket", ListenerConfig.builder()
                        .addProtocol(Http1Config.builder().continueImmediately(true).build())
                        .build())
```
Protocol configuration can now be done per socket.

**Comment by tomas-langer** [comment]
Fixed by #7008
